libdevice/include/device-sanitizer-report.hpp

```diff
@@ -0,0 +1,76 @@
+//==-- assert-happened.hpp - Structure and declaration for assert support --==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+// Treat this header as system one to workaround frontend's restriction
+#pragma clang system_header
+
+#include <cinttypes>
+
+#ifdef __SPIR__
+
+enum class DeviceSanitizerErrorType : int32_t {
+  UNKNOWN,
+  OUT_OF_BOUND,
+  MISALIGNED,
+  USE_AFTER_FREE,
+  OUT_OF_SHADOW_BOUND,
+};
+
+enum class DeviceSanitizerMemoryType : int32_t {
+  UNKNOWN,
+  USM_DEVICE,
```

Comment:
same as previous comment. I suggest to add a unified prefix like "DSM_" for all memory type.

Reply:
same as above.